samples/matter/common/src/bridge/bridge_shell.cpp

```diff
@@ -20,8 +21,17 @@ static int add_bridged_device(const struct shell *shell, size_t argc, char **arg
 		nodeLabel = argv[2];
 	}
 
-	CHIP_ERROR err =
-		GetBridgeManager().AddBridgedDevice(static_cast<BridgedDevice::DeviceType>(deviceType), nodeLabel);
+	auto *newBridgedDevice = BridgeFactory::GetBridgedDeviceFactory().Create(
+		static_cast<BridgedDevice::DeviceType>(deviceType), nodeLabel);
+	auto *newDataProvider = BridgeFactory::GetDataProviderFactory().Create(
+		static_cast<BridgedDevice::DeviceType>(deviceType), BridgeManager::HandleUpdate);
+
+	if (!newBridgedDevice || !newDataProvider) {
+		shell_fprintf(shell, SHELL_INFO, "Unsupported device type\n");
```

Comment:
I think that we may also fall into this case, if the memory could not be allocated using new, right? Maybe we should try to handle errors through some error code or print the generic error message without giving any details about the reason.

Reply:
Right, additional error handling would complicate the API, so I will just modify the error message.